src/components/NewCommentForm.tsx

```diff
@@ -1,8 +1,57 @@
-import React from 'react';
+import cn from 'classnames';
+import React, { FormEvent, useCallback, useState } from 'react';
+import { Comment, CommentData } from '../types/Comment';
+import { client } from '../utils/fetchClient';
+
+interface Props {
+  postId: number;
+  addComment: (newValue: Comment) => void;
+}
+
+export const NewCommentForm: React.FC<Props> = ({ postId, addComment }) => {
+  const [name, setName] = useState('');
+  const [email, setEmail] = useState('');
+  const [body, setBody] = useState('');
+  const [hasError, setHasError] = useState(false);
+  const [isSending, setIsSendign] = useState(false);
+
+  const isErrorVisible = useCallback((field: string) => (
+    !field.length && hasError
+  ), [hasError]);
+
+  const handleSubmit = (event: FormEvent) => {
+    event.preventDefault();
+
+    if (!name.length || !email.length || !body.length) {
+      setHasError(true);
+    } else {
+      const newComment = {
+        postId,
+        name,
+        email,
+        body,
+      };
+
+      setIsSendign(true);
+      client.post<CommentData>('/comments', newComment)
+        .finally(() => {
+          setIsSendign(false);
+          setBody('');
+        });
+
+      addComment({ id: Math.random() + postId, ...newComment });
```

Comment:
Submitted comments will have different ids on a server and in your app. You can reload comments from the server when new one was added